cmd/tdex/depositmarket.go

```diff
@@ -31,29 +60,553 @@ var depositmarket = cli.Command{
 	if err != nil {
 		return err
 	}
 	defer cleanup()
 
-	numOfAddresses := ctx.Int64("num_of_addresses")
-	resp, err := client.DepositMarket(
-		context.Background(),
-		&pboperator.DepositMarketRequest{
-			Market: &pbtypes.Market{
-				BaseAsset:  ctx.String("base_asset"),
-				QuoteAsset: ctx.String("quote_asset"),
+	net, baseAssetKey := getNetworkAndBaseAssetKey(ctx.String("network"))
+	fragmentationDisabled := ctx.Bool("no-fragment")
+	quoteAssetOpt := ctx.String("quote_asset")
+	baseAssetOpt := ""
+	if quoteAssetOpt != "" {
+		baseAssetOpt = baseAssetKey
+	}
+
+	if fragmentationDisabled {
+		numOfAddresses := int64(1)
+		if ctx.Int64("num_of_addresses") > 1 {
+			numOfAddresses = ctx.Int64("num_of_addresses")
+		}
+		resp, err := client.DepositMarket(
+			context.Background(), &pboperator.DepositMarketRequest{
+				Market: &pbtypes.Market{
+					BaseAsset:  baseAssetOpt,
+					QuoteAsset: quoteAssetOpt,
+				},
+				NumOfAddresses: numOfAddresses,
 			},
-			NumOfAddresses: numOfAddresses,
-		},
+		)
+		if err != nil {
+			return err
+		}
+
+		printRespJSON(resp)
+
+		return nil
+	}
+
+	randomWallet, err := trade.NewRandomWallet(&net)
+	if err != nil {
+		return err
+	}
+
+	log.Warnf("fund address: %v", randomWallet.Address())
+	explorerSvc, err := config.GetExplorer()
+	if err != nil {
+		log.WithError(err).Panic("error while setting up explorer service")
+	}
+	assetValuePair, unspents := findAssetsUnspents(
+		randomWallet,
+		explorerSvc,
+		baseAssetKey,
+	)
+
+	log.Info("calculating fragments ...")
+	baseFragments, quoteFragments := fragmentUnspents(assetValuePair)
+	outsLen := len(baseFragments) + len(quoteFragments)
+
+	log.Infof("base fragments: %v", baseFragments)
+	log.Infof("quote fragments: %v", quoteFragments)
+
+	feeAmount := wallet.EstimateTxSize(
+		len(unspents),
+		outsLen,
+		false,
+		MinMilliSatPerByte,
+	)
+
+	addresses, err := fetchMarketAddresses(
+		outsLen,
+		client,
+		baseAssetOpt,
+		quoteAssetOpt,
 	)
 	if err != nil {
 		return err
 	}
 
-	printRespJSON(resp)
+	outputs := createOutputs(
+		baseFragments,
+		quoteFragments,
+		feeAmount,
+		addresses,
+		assetValuePair,
+		baseAssetKey,
+	)
+
+	log.Info("crafting transaction ...")
+	inputBlindingKeys := [][]byte{
+		randomWallet.BlindingKey(),
+		randomWallet.BlindingKey(),
+	}
+	txHex, err := craftTransaction(
+		randomWallet,
+		unspents,
+		outputs,
+		feeAmount,
+		net,
+		baseAssetKey,
+		inputBlindingKeys,
+	)
+	if err != nil {
+		return err
+	}
+
+	log.Info("broadcasting transaction ...")
+
+	var txID string
+	for {
+		resp, err := explorerSvc.BroadcastTransaction(txHex)
+		if err != nil {
+			log.Warn(err)
+			log.Info("transaction broadcast retry")
+			continue
+		}
+		log.Info(resp)
+		txID = resp
+		break
+	}
+
+	outpoints := make([]*pboperator.TxOutpoint, 0, len(outputs))
+	for i := 0; i < len(outputs); i++ {
+		outpoints = append(outpoints, &pboperator.TxOutpoint{
+			Hash:  txID,
+			Index: int32(i),
+		})
+	}
```

Comment:
```suggestion
	outpoints := make([]*pboperator.TxOutpoint, outsLen, outsLen)
	for i:= 0; i < outsLen; i++ {
		outpoints[i] = &pboperator.TxOutpoint{
			Hash:  txID,
			Index: int32(i),
		})
	}
```

Reply:
We had this [discussion before](https://github.com/TDex-network/tdex-daemon/pull/319#discussion_r601619889), I dont think this is significant improvement if at all.
@tiero what do you think about this? should we use build in append or go with above way  of populating slice for known length?